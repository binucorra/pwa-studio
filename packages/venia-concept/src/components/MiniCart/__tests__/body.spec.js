import React from 'react';
import { createTestInstance } from '@magento/peregrine';

import Body from '../body';

jest.mock('../editItem');
jest.mock('../emptyMiniCart');
jest.mock('../productList');

jest.mock('src/components/LoadingIndicator', () => {
    return {
        __esModule: true,
        loadingIndicator: '( Loading Indicator Component Here )'
    };
});

const baseProps = {
    beginEditItem: jest.fn(),
    cart: {
        details: {},
        totals: {}
    }
};

test('renders the product list when appropriate', () => {
    const tree = createTestInstance(<Body {...baseProps} />).toJSON();

    expect(tree).toMatchSnapshot();
});

test('renders the loading indicator when appropriate', () => {
    const testProps = {
        ...baseProps,
        cart: {
            ...baseProps.cart,
            isLoading: true
        }
    };

    const tree = createTestInstance(<Body {...testProps} />).toJSON();

    expect(tree).toMatchSnapshot();
});

test('renders the empty mini cart when appropriate', () => {
    const testProps = {
        ...baseProps,
        isCartEmpty: true
    };

    const tree = createTestInstance(<Body {...testProps} />).toJSON();

    expect(tree).toMatchSnapshot();
});

test('renders the edit item dialog when appropriate', () => {
    const testProps = {
        ...baseProps,
        cart: {
            ...baseProps.cart,
            isEditingItem: true
        }
    };

    const tree = createTestInstance(<Body {...testProps} />).toJSON();

    expect(tree).toMatchSnapshot();
});
