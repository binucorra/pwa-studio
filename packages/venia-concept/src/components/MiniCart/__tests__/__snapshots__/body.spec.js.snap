// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the edit item dialog when appropriate 1`] = `"( Edit Item Component Here )"`;

exports[`renders the empty mini cart when appropriate 1`] = `"( Empty Mini Cart Component Here )"`;

exports[`renders the loading indicator when appropriate 1`] = `"( Loading Indicator Component Here )"`;

exports[`renders the product list when appropriate 1`] = `
<div>
  ( Product List Component Here )
</div>
`;
